File: adaptive-arp-api-csharp/Adaptive.Arp.Api/Adaptive/Arp/Api/Locale.cs
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     *
 *
 * =====================================================================================================================
 */

using Sharpen;

namespace Adaptive.Arp.Api
{
	/// <summary>Represents a specific user or system locate.</summary>
	/// <remarks>Represents a specific user or system locate.</remarks>
	public class Locale
	{
		/// <summary>A valid ISO Country Code.</summary>
		/// <remarks>A valid ISO Country Code.</remarks>
		/// <since>ARP1.0</since>
		private string country;

		/// <summary>A valid ISO Language Code.</summary>
		/// <remarks>A valid ISO Language Code.</remarks>
		/// <since>ARP1.0</since>
		private string language;

		/// <summary>Constructor used by the implementation</summary>
		/// <param name="country"></param>
		/// <param name="language"></param>
		/// <since>ARP1.0</since>
		public Locale(string country, string language)
		{
			this.country = country;
			this.language = language;
		}

		/// <summary>Returns the country code</summary>
		/// <returns>country code</returns>
		/// <since>ARP1.0</since>
		public virtual string GetCountry()
		{
			return country;
		}

		/// <summary>Set the country code</summary>
		/// <param name="country">code</param>
		/// <since>ARP1.0</since>
		public virtual void SetCountry(string country)
		{
			this.country = country;
		}

		/// <summary>Returns the language code</summary>
		/// <returns>language code</returns>
		/// <since>ARP1.0</since>
		public virtual string GetLanguage()
		{
			return language;
		}

		/// <summary>Set the language code</summary>
		/// <param name="language">code</param>
		/// <since>ARP1.0</since>
		public virtual void SetLanguage(string language)
		{
			this.language = language;
		}
	}
}

File: adaptive-arp-api-specs/src/me/adaptive/arp/api/Locale.java
/*
 * =| ADAPTIVE RUNTIME PLATFORM |=======================================================================================
 *
 * (C) Copyright 2013-2014 Carlos Lozano Diez t/a Adaptive.me <http://adaptive.me>.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations under the License.
 *
 * Original author:
 *
 *     * Carlos Lozano Diez
 *                 <http://github.com/carloslozano>
 *                 <http://twitter.com/adaptivecoder>
 *                 <mailto:carlos@adaptive.me>
 *
 * Contributors:
 *
 *     * Ferran Vila Conesa
 *                 <http://github.com/fnva>
 *                 <http://twitter.com/ferran_vila>
 *                 <mailto:ferran.vila.conesa@gmail.com>
 *
 *     * Aryslan
 *                 <http://github.com/Aryslan>
 *                 <http://twitter.com/Aryslan>
 *                 <mailto:ddbc@gft.com>
 *
 *     * Francisco Javier Martin Bueno
 *                 <https://github.com/kechis>
 *                 <mailto:kechis@gmail.com>
 *
 * =====================================================================================================================
 */

package me.adaptive.arp.api;

/**
 * Represents a specific user or system locate.
 *
 * @author Aryslan
 * @since ARP1.0
 */
public class Locale extends APIBean {

    /**
     * A valid ISO Country Code.
     */
    private String country;
    /**
     * A valid ISO Language Code.
     */
    private String language;

    /**
     * Default constructor
     *
     * @since ARP1.0
     */
    public Locale() {
    }

    /**
     * Constructor used by the implementation
     *
     * @param country  Country of the Locale
     * @param language Language of the Locale
     * @since ARP1.0
     */
    public Locale(String language, String country) {
        this.country = country;
        this.language = language;
    }

    /**
     * Returns the country code
     *
     * @return country code
     * @since ARP1.0
     */
    public String getCountry() {
        return country;
    }

    /**
     * Set the country code
     *
     * @param country code
     * @since ARP1.0
     */
    public void setCountry(String country) {
        this.country = country;
    }

    /**
     * Returns the language code
     *
     * @return language code
     * @since ARP1.0
     */
    public String getLanguage() {
        return language;
    }

    /**
     * Set the language code
     *
     * @param language code
     * @since ARP1.0
     */
    public void setLanguage(String language) {
        this.language = language;
    }
}
